internal/ethapi/tracer.go

```diff
@@ -240,21 +232,20 @@ func NewJavascriptTracer(code string, timeout time.Duration) (*JavascriptTracer,
 		db:         db,
 		dbvalue:    db.toValue(vm),
 		err:        nil,
-		timeout:    timeout,
 	}, nil
 }
 
+// Stop terminates execution of any JavaScript
+func (jst *JavascriptTracer) Stop(err error) {
+	jst.err = err
+	jst.vm.Interrupt <- func() {
+		panic(err)
+	}
+}
```

Comment:
> The assignment of jst.err is racy because it is also assigned in callSafely.

It is, but if the call simultaneously fails for two reasons, what matters is that at least one gets reported - I don't think there's a right answer as to which one.

> IMHO it would be simpler to just set the vm cancel channel to ctx.Done.

The VM cancel channel takes functions, so it's not type-compatible. This also wouldn't handle the case where someone calls cancel and the VM isn't currently executing.